src/DataTable/tableReducer.js

```diff
@@ -35,27 +35,28 @@ export function tableReducer(state, action) {
     }
 
     case 'SELECT_MULTIPLE_ROWS': {
-      const { selectedRows, rows } = action;
+      const { selectedRows, rows, mergeSelections } = action;
+      const selections = !mergeSelections ? selectedRows :
```

Comment:
[nit] ]This should just be 

```js
      const selections = mergeSelections 
             ? [...state.selectedRows, ...selectedRows.filter(row => !state.selectedRows.includes(row))];
             : selectedRows :
```

the negation may make this core harder to read.
        
```